tests/foreman/cli/test_content_credentials.py

```diff
@@ -0,0 +1,880 @@
+# -*- encoding: utf-8 -*-
+"""Test class for GPG Key CLI.
+The gpg sub-command was deprecated in favour of content-credential in
+Satellite 6.8
+
+:Requirement: ContentCredential
+
+:CaseAutomation: Automated
+
+:CaseLevel: Component
+
+:CaseComponent: ContentCredentials
+
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+from tempfile import mkstemp
+
+from fauxfactory import gen_alphanumeric
+from fauxfactory import gen_choice
+from fauxfactory import gen_integer
+from fauxfactory import gen_string
+
+from robottelo import ssh
+from robottelo.cli.base import CLIReturnCodeError
+from robottelo.cli.content_credentials import ContentCredential
+from robottelo.cli.factory import CLIFactoryError
+from robottelo.cli.factory import make_content_credential
+from robottelo.cli.factory import make_org
+from robottelo.cli.factory import make_product
+from robottelo.cli.factory import make_repository
+from robottelo.cli.org import Org
+from robottelo.cli.product import Product
+from robottelo.cli.repository import Repository
+from robottelo.constants import DEFAULT_ORG
+from robottelo.constants import VALID_GPG_KEY_FILE
+from robottelo.datafactory import invalid_values_list
+from robottelo.datafactory import valid_data_list
+from robottelo.decorators import tier1
+from robottelo.decorators import tier2
+from robottelo.decorators import upgrade
+from robottelo.helpers import get_data_file
+from robottelo.test import CLITestCase
+
+VALID_GPG_KEY_FILE_PATH = get_data_file(VALID_GPG_KEY_FILE)
+
+
+def create_gpg_key_file(content=None):
+    """Creates a fake GPG Key file and returns its path or None if an error
+    happens.
+    """
+
+    (_, key_filename) = mkstemp(text=True)
+    if not content:
+        content = gen_alphanumeric(gen_integer(20, 50))
+    with open(key_filename, "w") as gpg_key_file:
+        gpg_key_file.write(content)
+        return key_filename
+
+    return None
+
+
+class TestGPGKey(CLITestCase):
```

Comment:
Why write these based on unittest, instead of going straight to using pytest?  We'll have to refactor this, in short order, to use pytest.

Reply:
@mshriver I did not write them from scratch, they are clones of the gpg key tests. I wanted to get them approved as  quickly as possible so we have test coverage for 6.8 GA. No extra work is involved in my converting them later to not use unittest.